src/ui/public/agg_response/tabify/_buckets.js

```diff
@@ -34,5 +34,13 @@ export function AggResponseBucketsProvider() {
     }
   };
 
+  Buckets.prototype.reorder = function (params) {
```

Comment:
The reorder call creates a side-effect, fairly significant, and is only called from the tabify module. I'm not a 100% sure if this would have other impacts on the code, but it does break encapsulation somewhat.

I'd consider making the `reorder()` method return a clone of the buckets object, but with the keys ordered against the filters. So in `tabify` you'd have something like:

```
 if (buckets.length) {
    const reorderedBuckets = buckets.orderBucketsAccordingToFilterParams(agg.params);
    //... and then just work with reorderedBuckets...
 }
```